From b05fda0e5b8130e9ba8147caae358d579507b4ef Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:37:02 -0400
Subject: [misc] mei: me: set dma mask using DMA mapping API

Message-id: <1399041470-21330-30-git-send-email-prarit@redhat.com>
Patchwork-id: 79316
O-Subject: [RHEL7.1 PATCH BZ 1088953 29/77] mei: me: set dma mask using DMA mapping API
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit 3ecfb168a51ddf0226abe245065516bc2ebfdef5
Author: Tomas Winkler <tomas.winkler@intel.com>
Date:   Tue Dec 17 15:56:57 2013 +0200

    mei: me: set dma mask using DMA mapping API

    ME device is 64bit DMA capable
    We assume both coherent and consistent memory to match

    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/pci-me.c b/drivers/misc/mei/pci-me.c
index 49d34dd..1fc6e6e 100644
--- a/drivers/misc/mei/pci-me.c
+++ b/drivers/misc/mei/pci-me.c
@@ -170,6 +170,21 @@ static int mei_me_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
   dev_err(&pdev->dev, "failed to get pci regions.\n");
   goto disable_device;
  }
+
+ if (dma_set_mask(&pdev->dev, DMA_BIT_MASK(64)) ||
+     dma_set_coherent_mask(&pdev->dev, DMA_BIT_MASK(64))) {
+
+  err = dma_set_mask(&pdev->dev, DMA_BIT_MASK(32));
+  if (err)
+   err = dma_set_coherent_mask(&pdev->dev,
+          DMA_BIT_MASK(32));
+ }
+ if (err) {
+  dev_err(&pdev->dev, "No usable DMA configuration, aborting\n");
+  goto release_regions;
+ }
+
+
  /* allocates and initializes the mei dev structure */
  dev = mei_me_dev_init(pdev);
  if (!dev) {
-- 
1.7.1